front/app/components/ErrorToast/index.tsx

```diff
@@ -0,0 +1,155 @@
+import React, { useEffect, useRef, useState } from 'react';
+
+// components
+import {
+  Box,
+  Icon,
+  colors,
+  Text,
+  useBreakpoint,
+} from '@citizenlab/cl2-component-library';
+import CloseIconButton from 'components/UI/CloseIconButton';
+
+// utils
+import styled from 'styled-components';
+import eventEmitter from 'utils/eventEmitter';
+
+// events
+import { BUDGET_EXCEEDED_ERROR_EVENT } from 'components/ParticipationCTABars/VotingCTABar/events';
+import {
+  VOTES_EXCEEDED_ERROR_EVENT,
+  VOTES_PER_OPTION_EXCEEDED_ERROR_EVENT,
+} from 'components/AssignMultipleVotesControl';
+
+// intl
+import { useIntl } from 'utils/cl-intl';
+import messages from './messages';
+
+const StyledBox = styled(Box)`
+  opacity: 0;
+  z-index: 1000;
+  transition: all 0.6s ease-in-out;
+
+  &.visible {
+    opacity: 1;
+  }
+`;
+
+const ErrorToast = () => {
+  const atPageEnd = useRef(false);
+  const [error, setError] = useState<string | null>(null);
+  const [showError, setShowError] = useState(false);
+  const { formatMessage } = useIntl();
+  const [marginBottom, setMarginBottom] = useState('0px');
+  const isMobileView = useBreakpoint('phone');
+
+  // Listen for budgeting exceeded error
+  useEffect(() => {
+    const subscription = eventEmitter
+      .observeEvent(BUDGET_EXCEEDED_ERROR_EVENT)
+      .subscribe(() => {
+        setError(formatMessage(messages.budgetExceededError));
+        setShowError(true);
+      });
+
+    return () => {
+      subscription.unsubscribe();
+    };
+  }, [formatMessage]);
+
+  // Listen for voting per option exceeded error
+  useEffect(() => {
+    const subscription = eventEmitter
+      .observeEvent(VOTES_PER_OPTION_EXCEEDED_ERROR_EVENT)
+      .subscribe(() => {
+        setError(formatMessage(messages.votesPerOptionExceededError));
+        setShowError(true);
+      });
+
+    return () => {
+      subscription.unsubscribe();
+    };
+  }, [formatMessage]);
+
+  // Listen for voting exceeded error
+  useEffect(() => {
+    const subscription = eventEmitter
+      .observeEvent(VOTES_EXCEEDED_ERROR_EVENT)
+      .subscribe(() => {
+        setError(formatMessage(messages.votesExceededError));
+        setShowError(true);
+      });
+
+    return () => {
+      subscription.unsubscribe();
+    };
+  }, [formatMessage]);
+
+  const handleScroll = () => {
+    // Ref: https://stackoverflow.com/questions/63501757/check-if-user-reached-the-bottom-of-the-page-react
+    const bottom =
```

Comment:
The error needs to sit above the footer navigation on Desktop, so this listens for when that happens and sets the margin accordingly